src/coll_score/ucc_coll_score.h

```diff
@@ -134,9 +142,11 @@ ucc_status_t ucc_coll_score_build_map(ucc_coll_score_t *score,
 
 void         ucc_coll_score_free_map(ucc_score_map_t *map);
 
-/* Selects the "init" function from score map based on coll_args */
-ucc_status_t ucc_coll_score_map_lookup(ucc_score_map_t         *map,
-                                       ucc_base_coll_args_t    *args,
-                                       ucc_base_coll_init_fn_t *init,
-                                       ucc_base_team_t        **team);
+/* Initializes task based on args selection and score map.
+   Checks fallbacks if necessary. */
+ucc_status_t ucc_coll_score_map_init(ucc_score_map_t      *map,
+                                     ucc_base_coll_args_t *bargs,
+                                     ucc_coll_task_t     **task);
+
+ucc_status_t ucc_coll_task_post_with_fallback(ucc_coll_task_t *task);
```

Comment:
generic comment: it seems the whole score_map logic is getting complex. should we have a tool option to print all this information for debugging purposes?

Reply:
do you mean printing the score_map data struct, like active protocols selection? I also thought about that. Will try smth.